src/bootupd.rs

```diff
@@ -93,6 +93,27 @@ pub(crate) enum ComponentUpdateResult {
     },
 }
 
+fn ensure_writable_mount<P: AsRef<Path>>(p: P) -> Result<()> {
+    use nix::sys::statvfs;
+    let p = p.as_ref();
+    let stat = statvfs::statvfs(p)?;
+    if !stat.flags().contains(statvfs::FsFlags::ST_RDONLY) {
+        return Ok(());
+    }
+    let status = std::process::Command::new("mount")
+        .args(&["-o", "remount,rw"])
+        .arg(p.as_os_str())
```

Comment:
I think we can directly pass `p` here without `as_os_str()` since `Path` implements the `AsRef<OsStr>` trait right?